internal/fs/fs.go

```diff
@@ -219,6 +219,16 @@ func NewFileSystem(
 }
 
 func createFileCacheHandler(cfg *ServerConfig) (fileCacheHandler *file.CacheHandler, err error) {
+	// If parallel downloads are enabled then download-parallelism-per-file or
+	// max-download-parallelism or read-request-size-mb can't be 0.
+	if cfg.MountConfig.FileCacheConfig.EnableParallelDownloads {
```

Comment:
This and the enclosed "if" conditions can be combined. It'll reduce the indentation as well.